---
editor: 
  markdown: 
    wrap: 72
---

# Differential Gene Expression Analysis

::: callout-important
**Pre-reqs:** This code contains relative paths that only work within
the *raven* Roberts' Lab Server, where the data for this assignment is
housed
:::

#### 00-set-hooks
  This follows the instruction on how to automatically ignore large files from [THIS COURSE TUTORIAL](https://sr320.github.io/course-fish546-2023/assignments/02-DGE.html) , but there has to be a better way. It doesn't seem to work consistently. You may have to execute a line of code everytime you re-start your environment? Or something? Not fool-proof. 

#### 01-get-data
  Where TagSeq sample sequence fasta files from NCBI are downloaded and the Rutgers *Montipora capitata* reference genome V3 is downloaded

#### 02-align-hisat
  In-progress alignment of sample sequences to reference genome using HISAT
  
#### 04-create-kallisto-index
   Alignment of sample sequences to reference index using kallisto, however possibly not the appropriate tool, as kallisto is a 'pseudo-aligner' and HISAT should be used for organisms that have an annotated genome. Count matrix generated, but don't trust it?

#### 05-DESeq2
  DESeq2 differential gene expression analysis using dubious kallisto count matrix

#### 06-go-blast
Functional enrichment of WGCNA modules produced in `WGCNA_Mcap.Rmd` script for the *Montipora capitata* 2020 developmental time series.  This script uses the V3 genome from Cyanophora Rutgers.  


week05-presentation is published using Rpubs and can be found [HERE](http://rpubs.com/sarah_tanja/1034364).
